jicofo/rootfs/defaults/sip-communicator.properties

```diff
@@ -28,3 +28,11 @@ org.jitsi.jicofo.XMPP_MUC_COMPONENT_PREFIX={{ first (splitList "." .Env.XMPP_MUC
 {{ if and $ENABLE_AUTH (ne $AUTH_TYPE "jwt") }}
 org.jitsi.jicofo.auth.URL=XMPP:{{ .Env.XMPP_DOMAIN }}
 {{ end }}
+
+{{ $ENABLE_OCTO := .Env.TESTING_OCTO_PROBABILITY | default "0" | toBool }}
+{{ if $ENABLE_OCTO }}
+org.jitsi.jicofo.BridgeSelector.BRIDGE_SELECTION_STRATEGY={{ .Env.TESTING_OCTO_PROBABILITY }}
```

Comment:
Need to change `org.jitsi.jicofo.BridgeSelector.BRIDGE_SELECTION_STRATEGY={{ .Env.TESTING_OCTO_PROBABILITY }}` to `org.jitsi.jicofo.BridgeSelector.BRIDGE_SELECTION_STRATEGY={{ .Env. OCTO_BRIDGE_SELECTION_STRATEGY}}`

Reply:
Well 

> @goacid
> 
> I will mention the steps I have done for reference.
> 
> I have 4 virtual machines, with very high speed internet.
> Machine 	IP 	Role
> A 	172.16.4.200 	Full Docker Jitsi Meet Stack
> B 	172.16.4.231 	JVB1
> C 	172.16.4.232 	JVB2
> D 	172.16.4.233 	JVB3
> 
> For the OCTO configuration, I am using this PR
> 
> Using this, I have successfully setup my docker-jitsi-meet instance, listening on 443 with HTTP Redirected in Machine A. The .env file looks like this - https://pastebin.com/B2u6gJke
> 
> In the Machine **B, C** and **D**, I have cloned the same repo, and from what I understand, for the octo configuration we need to setup **only** _**jvb's**_ on different vm's or physical servers ( here B, C and D ) and connect it to the **prosody** of machine A. So my .env in these three machines has different passwords generated by **./gen-passwords** and all I have enabled **JVB_ENABLE_APIS=rest,colibri** in all three **.env** files. My docker-compose file for all these three machines contains configuration only for jvb which looks like this - https://pastebin.com/aGsygNwz
> ( P.S - I have **removed** the `depends_on: prosody` section, since only the container jvb will be running in all these machines. )
> 
> My questions are
> 
> **1.** Do I need to change the `JVB_AUTH_USER=jvb` setting in the .env file to something like
> jvb name 	Machine Role
> jvb1 	JVB1
> jvb2 	JVB2
> jvb3 	JVB3
> 
> and then run the jvb containers ?
> 
> **2.** How do I register these three jvb's from machine B,C and D to prosody of Machine A ? I believe it is done using the command below
> 
> `prosodyctl register jvb $DOMAIN $PASSWORD`
> 
> which in our case will be
> 
> `prosodyctl register jvb1 meet.jitsi $PASSWORD`
> `prosodyctl register jvb2 meet.jitsi $PASSWORD`
> `prosodyctl register jvb3 meet.jitsi $PASSWORD`
> 
> The $PASSWORD can be acquired from the respective .env file in machine B,C and D
> 
> **3.** After doing this, how do I go ahead ? How will the load-balancing happen ? Is there any step that I have missed ?
> 
> **4.** Correct me if I am wrong, when we add more video bridges ( not OCTO configuration ), jitsi loadbalances between the meetings
> 
> example. Meeting A will be scheduled on jvb1, Meeting B will be on jvb2
> 
> But using OCTO, participants of the same meeting can be accommodated on different jvb's which results in increased participant capacity for the same meeting >75
> 
> **5.** Also in case of a single machine, lets say my Machine A has 128 CPU's, 3 Ti of RAM , how do I setup octo for a single machine ? 4 jvb's on a machine A with OCTO setup
> 
> Let me know if you need any more information.

Let me explain how is it doing : 
JVB's use the same username/password, no need to create different ones.
This is 'by design' when jvb configuration files are generate when multiple JVB's machine on the docker install. Sharding is 'out of the box'.

So it should work out of the box IF you use different JVB hosts, just be sur that 'docker stack' can communicate with JVB host (docker or not)
If you use same host for multiple JVB container, then it is no more out of the box. Because of listen ports (you need one jvb/octo ports per container, those ports can't be the same on multiple jvb containers).

Generation of jvb conf you have : 
`{{ $ENABLE_OCTO := .Env.TESTING_OCTO_PROBABILITY | default "0" | toBool }}
{{ if $ENABLE_OCTO }}
org.jitsi.videobridge.octo.BIND_ADDRESS={{ .Env.JVB_NAT_HARVESTER_LOCAL_ADDRESS }}
org.jitsi.videobridge.octo.PUBLIC_ADDRESS={{ .Env.JVB_NAT_HARVESTER_PUBLIC_ADDRESS }}
org.jitsi.videobridge.octo.BIND_PORT={{ .Env.OCTO_BIND_PORT | default "4096" }}
org.jitsi.videobridge.REGION={{ .Env.OCTO_REGION | default "Region1" }}
{{ end }}`
So OCTO_BIND_PORT should be different if container are on same docker host ! If not you can use same .env file.
Same behaviour for JVB_PORT (udp listen port of jvb) by the way.

So your JVB must be able to reach prosody and jicofo AND each others on OCTO_BIND_PORT (all docker container and JVB must be able to communicate)

When your JVBs are setup this way, they must register in prosody during startup.

Then comes the conf part of JICOFO.
`{{ $ENABLE_OCTO := .Env.TESTING_OCTO_PROBABILITY | default "0" | toBool }}
{{ if $ENABLE_OCTO }}
org.jitsi.jicofo.BridgeSelector.BRIDGE_SELECTION_STRATEGY={{ .Env.TESTING_OCTO_PROBABILITY }}
{{ end }}`
As you can see here you choose the mode of the 'loadbalancing'.
OCTO must be setup in Jicofo !

Regarding number of participants, I success to torture a 2 octo JVB setup with 70 users, BUT the client (chrome/firefox/chromium whatever) use to loose packet ... another issue (i began a rex [here](https://community.jitsi.org/t/jitsi-max-users-in-same-room/81281)